### PR TITLE
perf: pathology - Update tile inference CLI tool

### PR DESCRIPTION
**Pull request by aauker**
Performance and usage improvement for inference CLI.  This tool pulls models from those specified in https://github.com/msk-mind/luna-ml/blob/main/hubconf.py

The output format is the same, but with the class labels correctly propagated if given as model attributes.

Arguments can be specified via CLI or json/yaml, with CLI args overriding.  All necessary arguments are saved in the output directory for reproducibility. 

Usage docs:
```
Usage: infer_tile_labels.py [OPTIONS]

  Run with explicit arguments:

      infer_tiles
          -i 1412934/data/TileImages
          -o 1412934/data/TilePredictions
          -r msk-mind/luna-ml:main 
          -t tissue_tile_net_transform 
          -m tissue_tile_net_model_5_class
          -w main:tissue_net_2021-01-19_21.05.24-e17.pth

  Run with implicit arguments:

      infer_tiles -m 1412934/data/TilePredictions/metadata.json

  Run with mixed arguments (CLI args override yaml/json arguments):

      infer_tiles --input_data 1412934/data/TileImages -m 1412934/data/TilePredictions/metadata.json

Options:
  -i, --input_data TEXT         path to input data
  -o, --output_dir TEXT         path to output directory to save results
  -rn, --repo_name TEXT         repository name to pull model and weight from,
                                e.g. msk-mind/luna-ml

  -tn, --transform_name TEXT    torch hub transform name
  -mn, --model_name TEXT        torch hub model name
  -wt, --weight_tag TEXT        weight tag filename
  -nc, --num_cores INTEGER      Number of cores to use
  -bx, --batch_size INTEGER     weight tag filename
  -m, --method_param_path TEXT  json file with method parameters for tile
                                generation and filtering

  --help                        Show this message and exit.

```

### CODE REVIEW

File: pyluna-pathology/luna/pathology/cli/infer_tile_labels.py [hunk 1]
```diff
@@ -1,95 +1,148 @@
...
+    Run with explicit arguments:
+
+    \b
+        infer_tile_labels
+            -i 1412934/data/TileImages
+            -o 1412934/data/TilePredictions
+            -r msk-mind/luna-ml:main 
+            -t tissue_tile_net_transform 
+            -m tissue_tile_net_model_5_class
+            -w main:tissue_net_2021-01-19_21.05.24-e17.pth
```
Comment (doori) [inline pyluna-pathology/luna/pathology/cli/infer_tile_labels.py:51]: Update the flags to `-rn` `-wt` etc. Is `\b` part of the docstring?

Comment (aauker) [reply pyluna-pathology/luna/pathology/cli/infer_tile_labels.py:51]: The \b preserves the indenting of the text block when showing the CLI help method

File: pyluna-pathology/luna/pathology/cli/infer_tile_labels.py [hunk 2]
```diff
@@ -1,95 +1,148 @@
...
+
+    Run with implicit arguments:
+
+    \b
+        infer_tile_labels -m 1412934/data/TilePredictions/metadata.json
+    
+    Run with mixed arguments (CLI args override yaml/json arguments):
+    
+    \b
+        infer_tile_labels --input_data 1412934/data/TileImages -m 1412934/data/TilePredictions/metadata.json
```
Comment (doori) [inline pyluna-pathology/luna/pathology/cli/infer_tile_labels.py:61]: we can add `infer_tile_labels` as an entrypoint in pyluna-pathology/setup.cfg

**Review by doori — APPROVED** [review]
Great improvements in inference CLI and passing/validating the parameters!!

**Pull request merged**